src/linux/mod.rs

```diff
@@ -85,3 +92,99 @@ impl Sandbox for LinuxSandbox {
         }
     }
 }
+
+/// Restrict networking using seccomp and namespaces.
+fn restrict_networking() -> Result<()> {
+    // Create network namespace.
+    let result = create_user_namespace(true).and_then(|_| unshare(Namespaces::NETWORK));
+
+    // Apply seccomp network filter.
+    match NetworkFilter::apply() {
+        _ if result.is_ok() => Ok(()),
+        result => result,
+    }
+}
+
+/// Create a new user namespace.
+///
+/// If the `become_root` flag is set, then the current user will be mapped to
+/// UID 0 inside the namespace. Otherwise the current user will be mapped to its
+/// UID of the parent namespace.
+fn create_user_namespace(become_root: bool) -> Result<()> {
+    // Get the current UID.
+    let uid = unsafe { libc::getuid() };
+
+    // Create the namespace.
+    unshare(Namespaces::USER)?;
+
+    // Map the UID and GID.
+    let map = if become_root { format!("0 {uid} 1\n") } else { format!("{uid} {uid} 1\n") };
+    fs::write("/proc/self/uid_map", map.as_bytes())?;
```

Comment:
Looking through [`man 7 user_namespaces`][1], there is a set of restrictions for writing to this file if you do not have `CAP_SETUID`. That includes:

> The data written to uid_map (gid_map) must consist of a single line that maps the writing process's effective user ID (group ID) in the parent user namespace to a user ID (group ID) in the user namespace.

[1]: https://man7.org/linux/man-pages/man7/user_namespaces.7.html

So should this be calling `libc::geteuid()` for the effective UID and not `libc::getuid()`.

Reply:
That's what `unshare -U` does too, so it makes sense.